eventuals/collect.h

```diff
@@ -5,45 +5,70 @@
 #include "eventuals/loop.h"
 #include "eventuals/type-traits.h"
 
-// NOTE: The following include is a rather large one.
-#include "google/protobuf/repeated_field.h"
-
 ////////////////////////////////////////////////////////////////////////
 
 namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename Container>
+template <typename Collection, typename Enable = void>
+struct EventualsCollector {
+  template <typename T>
+  void Collect(Collection& collection, T&& value) {
+    static_assert(
+        !std::is_same_v<void, Enable>,
+        "Provide your own EventualsCollector");
+  }
+};
+
+////////////////////////////////////////////////////////////////////////
+
+// Collectors for STL collections.
+
+template <typename Collection>
+struct EventualsCollector<
+    Collection,
+    std::enable_if_t<
+        eventuals::HasEmplaceBack<Collection>::value>> {
+  template <typename T>
+  std::enable_if_t<
+      std::is_convertible_v<
+          T,
+          typename Collection::value_type>>
```

Comment:
I love this checking this constraint, but are we trying to "select" vis SFINAE here or just providing better error messages? It seems like the latter, as in, we've already selected via `EventualsCollector<C>`, and now we just want better error messages, so let's use a `static_assert()` in the body instead! Here and below please!

Reply:
Yeah, you're right x)

Let me fix that..